073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,250 @@
+2. Create the pods for the cluster and add the cluster-ca-cert-generation and cluster-ca-key-generation annotations, set to the current values passed in with the Cluster CA.
+
+EntityOperatorReconciler, CruiseControlReconciler and KafkaExporterReconciler generate the certificates they will use to authenticate with Kafka and ZooKeeper and store them in the Kubernetes Secrets listed above, with the cluster-ca-cert-generation set to the current value passed in with the Cluster CA.
+
+### Example flows
+
+When a new cluster is first created the flow is:
+
+1. Either user or CaReconciler creates the Cluster and Client CA certificates and keys and stores them in Kubernetes Secrets
+2. The CaReconciler generates the Cluster Operator certificate and stores it in the <CLUSTER_NAME>-cluster-operator-certs Kubernetes Secret
+3. The component reconcilers (KafkaReconciler, ZooKeeperReconciler, EntityOperatorReconciler etc) generate their certificates, store them in Kubernetes Secrets and annotate the Kubernetes Secrets and Pods with the cert and key generations
+
+When the CA private key is updated (either because it has expired or the user has incremented the generation to indicate they've manually updated the Kubernetes Secret):
+1. If Strimzi is managing the CA, the CaReconciler renews the Cluster and Client CAs certificates and keys. The old certificate and key are kept in the Kubernetes Secret and renamed to have the key ca-YYYY-MM-DDTHH-MM-SSZ.crt.
+2. The CaReconciler skips updating the <CLUSTER_NAME>-cluster-operator-certs Kubernetes Secret (since the components still only trust the old CA)
+3. The CaReconciler rolls the ZooKeeper and Kafka nodes to trust the new CA certificates
+4. The ZooKeeper and Kafka reconcilers generate new certificates signed by the new CA key and roll the pods to use the new certificates
+5. The EntityOperator, CruiseControl and KafkaExporter reconcilers generate new certificates signed by the new CA key amd roll the pods to use the new certificates
+6. A new reconciliation starts
+7. The CaReconciler updates the <CLUSTER_NAME>-cluster-operator-certs Kubernetes Secret with the new CA certificate (since the generations are now correct)
+8. The CaReconciler removes the old CA certificates and keys that have been named ca-YYYY-MM-DDTHH-MM-SSZ.crt.
+
+The existing logic assumes two things:
+ - all certificates used by components have been signed by the same root CA certificate
+ - the CaReconciler runs and (if required) has either created or updated the CA certificates and keys before the other reconcilers run, i.e. the CA certificate is already present when e.g. the KafkaReconciler starts
+
+## Motivation
+
+In future it would be beneficial if Strimzi can integrate with commonly deployed certificate management solutions, such as [Cert Manager][cmio], or [Vault][vault].
+These solutions work by having the user request certificates from a particular issuer.
```

Comment:
Maybe we can use something better than _user_? As that suggests it has to be done manually which is not what we  in general want / expect. Can we make it more neutral as to who requests it? User, other system, etc.